packages/react-day-picker/src/hooks/useDayRender/useDayRender.tsx

```diff
@@ -107,7 +107,7 @@ export function useDayRender(
   const buttonProps = {
     ...divProps,
     disabled: activeModifiers.disabled,
-    role: 'gridcell',
+    role: 'button',
```

Comment:
I believe this will break accessibility in some ways because `button` elements are not supposed to be children of `grid > rowgroup`. I would recommend testing any changes with a screen reader.

I've found that the structure that makes the most sense is to add a `div` with the role of `gridcell` and the `button` inside the `gridcell`. Additionally, using the `aria-pressed` role for the button when it is selected (since `aria-selected` is not valid attribute for `button` elements, and simply having `aria-selected` in the `gridcell` did not work in VoiceOver. (as in, it didn't announce that the day was selected).

So

```html
<table role="grid">
<!-- OMITTED HEAD -->
    <tbody role="rowgroup">
      <tr>
        <td role="presentation">
          <div role="gridcell">
            <button name="day" aria-label="April 1st, 2022" aria-pressed="false"  tabindex="0" type="button">1</button>
           </div>
        </td>
        <!-- !! SELECTED STATE -->
        <td role="presentation">
          <div role="gridcell" aria-selected="true">
            <button name="day" aria-label="April 1st, 2022" aria-pressed="true"  tabindex="0" type="button">1</button>
           </div>
        </td>
      </tr>
   </tbody>
</table>
```

Reply:
You can possibly consider removing the `div` and changing `role="presentation"` to `role="gridcell"` in the `td` element. But please test in a screen reader.

Reply:
🤔 I have doubts about this approach. From my research it seems `gridcell` elements are grouping elements (except when the content is editable [think spreadsheet, for example]). In which case, clicking on the cells is just used to focus on the cell to allow the user to type. The way in which the `DayPicker` works, each Day behaves as a toggle button. Clicking on it, toggles the selected/unselected state. A `gridcell` by itself wont be able to communicate the sort of interactions each Day supports, as well as a toggle button can. 

I also looked at the ARIA role hierarchy to support this point, where you can see interactive elements generally stem from the abstract `widget` role. There you'll see the `button` role, whereas the `gridcell` role inherits from `section`, which is a `structure` role, and is mainly for grouping content.

Resources: 

- [interactive `gridcell`'](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/gridcell_role#interactive_grids_and_treegrids)
- [ARIA role hierarchy](https://www.w3.org/TR/wai-aria-1.0/rdf_model.svg)

Reply:
Additionally you can consider making each day a `checkbox` or `radio` depending on the `mode` (`single` -> `radio`; `multiple` -> `checkbox`) 💡